### PR TITLE
Protect against missing overrides section

### PR DESCRIPTION
**Pull request by joshmyers**
We are using grafonnet-lib to generate dashboards. These dashboards do
not contain an `override` key in `fieldConfig` by default and that is
causing this DashboardMigrator script to blow up when trying to import
the dashboards, see [1]. Using grafonnet-lib, it isn't possible to set an
empty overrides attribute e.g. [2] requires matcher/properties to be
set. Setting to null ends up giving me [3], which causes the panel to
not be displayed.

[1]
```
initDashboard.ts:137 TypeError: t.overrides is not iterable
    at v.w (DashboardMigrator.ts:960)
    at v.updateSchema (DashboardMigrator.ts:672)
    at D.updateSchema (DashboardModel.ts:993)
    at new D (DashboardModel.ts:156)
    at initDashboard.ts:134
```

[2]
https://github.com/grafana/grafonnet-lib/blob/master/grafonnet/stat_panel.libsonnet#L150-L164

[3]
```
"fieldConfig": {
   "defaults": {
      "links": [ ],
      "mappings": [ ],
      "thresholds": {
         "mode": "absolute",
         "steps": [
            {
               "color": "red",
               "value": 0
            },
            {
               "color": "orange",
               "value": 1
            },
            {
               "color": "green",
               "value": 3
            }
         ]
      },
      "unit": "none"
   },
   "overrides": [
      { }
   ]
},
```

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: public/app/features/dashboard/state/DashboardMigrator.ts [hunk 1]
```diff
@@ -957,10 +957,13 @@ function upgradeValueMappingsForPanel(panel: PanelModel) {
 
   fieldConfig.defaults.mappings = upgradeValueMappings(fieldConfig.defaults.mappings, fieldConfig.defaults.thresholds);
 
-  for (const override of fieldConfig.overrides) {
-    for (const prop of override.properties) {
-      if (prop.id === 'mappings') {
-        prop.value = upgradeValueMappings(prop.value);
+  // Protect against no overrides
+  if (fieldConfig.hasOwnProperty('overrides')) {
```
Comment (mckn) [inline public/app/features/dashboard/state/DashboardMigrator.ts:961]:
```suggestion
  if (Array.isArray(fieldConfig.overrides)) {
```

**Review by mckn — COMMENTED** [review]
Thanks for fixing this! I added a comment on how you could improve the check a little bit to cater for a scenario where someone has set the overrides to a non-array value.

**Review by dprokop — APPROVED** [review]
Looks good, just please merge @mckn's suggestion :)

**Comment by joshmyers** [comment]
Thanks @mckn and @dprokop - done!

**Pull request merged**